va/va_test.go

```diff
@@ -734,3 +736,434 @@ func TestLogRemoteDifferentials(t *testing.T) {
 		})
 	}
 }
+
+// setup returns an in-memory VA and a mock logger. The default resolver client
+// is MockClient{}, but can be overridden.
+func setupVA(srv *httptest.Server, ua string, rvas []RemoteVA, mockDNSClient bdns.Client) (*ValidationAuthorityImpl, *blog.Mock) {
```

Comment:
This function looks like a copy of `setup()`, which is still retained in this file. I copied the `setup()` from the `main` branch into `old.txt` and diffed it against `setupVA()` on this branch.

```diff
--- /tmp/old.txt	2024-11-15 16:58:07.334618542 -0800
+++ /tmp/new.txt	2024-11-15 16:57:49.670734507 -0800
@@ -1,29 +1,28 @@
-func setup(srv *httptest.Server, maxRemoteFailures int, userAgent string, remoteVAs []RemoteVA, mockDNSClientOverride bdns.Client) (*ValidationAuthorityImpl, *blog.Mock) {
+func setupVA(srv *httptest.Server, ua string, rvas []RemoteVA, mockDNSClient bdns.Client) (*ValidationAuthorityImpl, *blog.Mock) {
 	features.Reset()
 	fc := clock.NewFake()
 
-	logger := blog.NewMock()
-
-	if userAgent == "" {
-		userAgent = "user agent 1.0"
+	mockLog := blog.NewMock()
+	if ua == "" {
+		ua = "user agent 1.0"
 	}
 
 	va, err := NewValidationAuthorityImpl(
-		&bdns.MockClient{Log: logger},
+		&bdns.MockClient{Log: mockLog},
 		nil,
-		maxRemoteFailures,
-		userAgent,
+		0,
+		ua,
 		"letsencrypt.org",
 		metrics.NoopRegisterer,
 		fc,
-		logger,
+		mockLog,
 		accountURIPrefixes,
 		PrimaryPerspective,
-		"",
+		"ARIN",
 	)
 
-	if mockDNSClientOverride != nil {
-		va.dnsClient = mockDNSClientOverride
+	if mockDNSClient != nil {
+		va.dnsClient = mockDNSClient
 	}
 
 	// Adjusting industry regulated ACME challenge port settings is fine during
@@ -37,8 +36,8 @@
 	if err != nil {
 		panic(fmt.Sprintf("Failed to create validation authority: %v", err))
 	}
-	if remoteVAs != nil {
-		va.remoteVAs = remoteVAs
+	if rvas != nil {
+		va.remoteVAs = rvas
 	}
-	return va, logger
-}
+	return va, mockLog
+}
```

It looks like most of the changes here are renames. There's one substantive change: `setupVA`  doesn't have a `maxRemoteValidations` parameter, but `setup` does. Turns out I actually made the same change to `setup` in #7810! Looks like we were thinking along the same lines. Given that, I think this diff can be reverted, and the new tests can just use `setup()`.

Oh, and the other substantive change: making RIR unconditionally be `ARIN`. In https://github.com/letsencrypt/boulder/pull/7815/files#diff-38dce5cfdf5ee8afeb3994e69b1d3663bbd6ebf4067342bb175312854dfaf0cdR119-R124 I did something similar with `perspective`, with two differences: I set it to PrimaryPerspective if there were remoteVAs, and otherwise set it to a random value. I think we should do the same thing with RIR. It should be empty when simulating the primary, and a random value when simulating a remoteVA. That should work okay for general-purpose tests. I see below that for MPIC-specific tests, setupRVAs manually sets the field based on a test case, so we'll still get the behavior we want in terms of being able to set two instances of the same RIR.

For the renames: They don't seem critical to this PR, so my general advice is to not include them since they're distracting. If they're important, let's do them in another PR like we did for the other renames.

Reply:
I was under the impression that we were porting all of this back into PerformValidation. Lets chat about this PR when you get in today.